src/gkmodel/node_model.py

```diff
@@ -232,17 +236,32 @@ def predict(self,
             for study_id in data.study_id
         ])
 
+        adjust_values = np.zeros(self.data.num_points)
+
```

Comment:
maybe also add `data._sort_by_data_id()`, to make sure the order :)

Reply:
is data sorted in `create_design_mat`?

Reply:
I don't think so :) Since it might be good to only sort after the fitting process.

Reply:
so I sort before the `create_design_mat` part? We want `mat` and `soln` to have the same order right?